module/msg_smt/src/mod_msg_smt.c

```diff
@@ -0,0 +1,477 @@
+
+    /* Driver entity identifier */
+    fwk_id_t driver_id;
+
+    /* SCMI module service bound to the channel */
+    fwk_id_t scmi_service_id;
+
+    /* Driver API */
+    struct mod_msg_smt_driver_ouput_api *driver_api;
+
+    /* SCMI service API */
+    struct mod_scmi_from_transport_api *scmi_api;
+
+    /* Flag indicating the mailbox is ready */
+    bool msg_smt_mailbox_ready;
+};
+
+struct smt_ctx {
+    /* Table of channel contexts */
+    struct smt_channel_ctx *channel_ctx_table;
+
+    /* Number of channels */
+    unsigned int channel_count;
+};
+
+static struct smt_ctx smt_ctx;
+
+/*
+ * SCMI Transport API
+ */
+static int smt_get_secure(fwk_id_t channel_id, bool *secure)
+{
+    if (secure == NULL) {
+        assert(false);
+        return FWK_E_PARAM;
+    }
+
+    /* No msg mailbox is considered secure */
+    *secure = 0;
+
+    return FWK_SUCCESS;
+}
+
+static int smt_get_max_payload_size(fwk_id_t channel_id, size_t *size)
+{
+    struct smt_channel_ctx *channel_ctx;
+
+    if (size == NULL) {
+        assert(false);
+        return FWK_E_PARAM;
+    }
+
+    channel_ctx =
+        &smt_ctx.channel_ctx_table[fwk_id_get_element_idx(channel_id)];
+
+    *size = channel_ctx->max_payload_size;
+
+    return FWK_SUCCESS;
+}
+
+static int smt_get_message_header(fwk_id_t channel_id, uint32_t *header)
+{
+    struct smt_channel_ctx *channel_ctx;
+
+    if (header == NULL) {
+        assert(false);
+        return FWK_E_PARAM;
+    }
+
+    channel_ctx =
+        &smt_ctx.channel_ctx_table[fwk_id_get_element_idx(channel_id)];
+
+    if (!channel_ctx->locked)
+        return FWK_E_ACCESS;
+
+    *header = channel_ctx->in->message_header;
+
+    return FWK_SUCCESS;
+}
+
+static int smt_get_payload(fwk_id_t channel_id,
+                           const void **payload,
+                           size_t *size)
+{
+    struct smt_channel_ctx *channel_ctx;
+
+    if (payload == NULL) {
+        assert(false);
+        return FWK_E_PARAM;
+    }
+
+    channel_ctx =
+        &smt_ctx.channel_ctx_table[fwk_id_get_element_idx(channel_id)];
+
+    if (!channel_ctx->locked)
+        return FWK_E_ACCESS;
+
+    *payload = channel_ctx->in->payload;
+
+    if (size != NULL) {
+        *size = channel_ctx->in_len -
+            sizeof(channel_ctx->in->message_header);
+    }
+
+    return FWK_SUCCESS;
+}
+
+static int smt_write_payload(fwk_id_t channel_id,
+                             size_t offset,
+                             const void *payload,
+                             size_t size)
+{
+    struct smt_channel_ctx *channel_ctx;
+
+    channel_ctx =
+        &smt_ctx.channel_ctx_table[fwk_id_get_element_idx(channel_id)];
+
+    if ((payload == NULL)                         ||
+        (offset  > channel_ctx->max_payload_size) ||
+        (size > channel_ctx->max_payload_size)    ||
+        ((offset + size) > channel_ctx->max_payload_size)) {
+
+        assert(false);
+        return FWK_E_PARAM;
+    }
+
+    if (!channel_ctx->locked)
```

Comment:
all single-line if conditions should be rounded by braces {}

Reply:
yes, I missed this one